src/Monitor.ts

```diff
@@ -64,11 +85,19 @@ class Monitor {
 
         if (options?.shouldMonitorError?.(err) ?? true) {
             this.plugins.onFailure({scope, executionTime, options, reason: err});
+
+            const {context, logAsError, logErrorAsInfo} = options;
+            this.logger.error(`${scope}.error`, err, context, logAsError, logErrorAsInfo);
         }
 
         throw err;
     }
 
+    monitoredLogger = (level: MonitoredOptions<any>['level'], message, {extra}) => {
+        const log = level === 'info' ? this.logger.info : this.logger.debug;
+        log(message, {extra});
+    };
+
     getStatsdClient = () => undefined;
```

Comment:
I'd put a `@deprecated` JSDoc comment over `getStatsdClient`, here and in the global exports